src/style/style.js

```diff
@@ -1365,6 +1365,10 @@ class Style extends Evented {
         this.light.updateTransitions(parameters);
     }
 
+    getTerrain() {
+        return this.terrain ? this.terrain.get() : this.stylesheet.terrain;
```

Comment:
No that's unnecessary, we only need the following:

```suggestion
        return this.terrain ? this.terrain.get() : null;
```

If it's in the stylesheet, this.terrain will get itself initialized.